src/main/java/hudson/remoting/RemoteClassLoader.java

```diff
@@ -1124,6 +1121,88 @@ private Object readResolve() throws ObjectStreamException {
         }
     }
 
+    /**
+     * An {@link InvocationHandler} for proxying calls to a target {@link IClassLoader} which catches
+     * {@link RemotingSystemException} caused by an {@link InterruptedException} (or {@link InterruptedIOException}).
+     * When this happens, the target method is called again, until it successfully returns (or throws a different kind
+     * of exception), or a maximum number of retries has been reached. The current thread is then interrupted again.
```

Comment:
nit: 

Maybe something like this:
```suggestion
     * When this happens, the target method is called again, until it successfully returns, throws a different kind
     * of exception, or a maximum number of retries has been reached. The current thread is then interrupted again.
```

Reply:
505ad3b4